### DISCUSSION TITLE
Is there a way to disable authentication persistence?

### DISCUSSION THREAD

**Discussion by aureliopetrone** [invertase/react-native-firebase#6729] [Q&A]
Hello guys, Is there a way to disable authentication persistence for both iOS/Android?

**Comment by mikehardy** [comment]
Hi there! This sounds security-related and in my experience any question that may be security related should be a) as detailed as possible (*exactly* what do you mean by persistence? What is it exactly that you assert is being persisted that you do not want to persist, and where is it being persisted?) and it should be b) asked in the authoritative repositories - that is you should ask upstream in firebase-ios-sdk and firebase-android-sdk repositories as they will be able to discuss/answer authoritatively, whereas I would be a poor facade for their actual answers

**Comment by Shakahs** [comment]
https://firebase.google.com/docs/auth/web/auth-state-persistence

`firebase.auth.Auth.Persistence.NONE | 'none' | Indicates that the state will only be stored in memory and will be cleared       when the window or activity is refreshed.`

Disabling authentication persistence is desirable for testing, so the app forgets the user on exit. I'm trying to test on iOS with Detox and the authentication state persists even with Detox uninstalling and reinstalling the app. 

Firebase docs describe this API, and specifically mention React Native, but I'm not able to find it in RNFirebase.

**Reply by mikehardy** [reply]
Why not just on startup `firebase().auth.currentUser && await firebase().auth.currentUser.logOut()` (with apologies, haven't checked the actual API but it should be something like that)

I'm not aware of this API existing in react-native-firebase
